misc/dbt-materialize/dbt/include/materialize/macros/utils/cast.sql

```diff
@@ -0,0 +1,27 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License in the LICENSE file at the
+-- root of this repository, or online at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+{% macro materialize__cast(expression, data_type) -%}
+    {#-- Handle types that don't support cast(NULL as type) --#}
+    {%- if expression.strip().lower() == "null" and data_type.strip().lower() == "map" -%}
+      MAP[NULL=>NULL]
+    {%- elif expression.strip().lower() == "null" and data_type.strip().lower() == "list" -%}
+      LIST[NULL]
+    {%- elif expression.strip().lower() == "null" and data_type.strip().lower() == "record" -%}
+      ROW(NULL)
```

Comment:
```suggestion
      (select row() where false)
```

There isn't a better way to create a `NULL` of type `record` that I'm aware of :/.